0000-template.md

```diff
@@ -1,51 +1,48 @@
+- Title: (fill me in with a human-focussed title for the RFC)
 - Feature Name: (fill me in with a unique ident, `my_awesome_feature`)
 - Start Date: (fill me in with today's date, YYYY-MM-DD)
-- RFC PR: [rust-lang/rfcs#0000](https://github.com/rust-lang/rfcs/pull/0000)
-- Rust Issue: [rust-lang/rust#0000](https://github.com/rust-lang/rust/issues/0000)
+- RFC PR: [#0000](https://github.com/rust-lang/rfcs/pull/0000)
+- Tracking Issue: [rust-lang/rust#0000](https://github.com/rust-lang/rust/issues/0000)
+- Team: (fill me in with the team or teams responsible for this RFC)
+- Keywords: (fill me in with keywords which may be useful when searching for this RFC)
+- Previous RFCs: [#0000](https://github.com/rust-lang/rfcs/pull/0000) (fill me in with RFC numbers for any RFCs this RFC supersedes, deprecates, or extends)
+- Previous discussion: (fill me in with links to previous discussions such as internals.rust-lang.org or Zulip threads, or issues)
 
 # Summary
 [summary]: #summary
 
 One paragraph explanation of the feature.
 
-# Motivation
-[motivation]: #motivation
+# Motivation and background
+[motivation-and-background]: #motivation-and-background
 
-Why are we doing this? What use cases does it support? What is the expected outcome?
+Why are we doing this? What use cases does it support? What is the expected outcome? Why are existing solutions not good enough? If possible, include data to support your claims. Include any background context useful for understanding the RFC.
```

Comment:
I think "Why are existing solutions not good enough" should go in the alternatives section, not here.

"include data" is a higher bar than we have ever set for RFCs; by all means if someone *has* data they should include it, but for many RFCs there may not be a meaningful way to collect data short of a full user study.

Reply:
The point I was trying to make is to discuss why *any* new feature is required, rather than why this proposal is better than an alternative (I guess 'do nothing' is always an alternative, but it always seems a bit forced).

The request for data is specially prefixed with "If possible" because it is a high bar and I hope makes clear that is not required, but I think suggesting it is useful as a way to at least make people think in concrete terms where possible rather than 'just vibes' type motivations.

Reply:
The existing "rationale and alternatives" section includes "- What is the impact of not doing this?", and I think that's been useful; I think explicitly asking for discussion of the null alternative would help even more. I don't think it's "forced", I think it's genuinely useful to talk about what happens if we don't make the change. It also goes along with things like "could this be done as a macro", or "could this language change be done as a library feature", or "could this standard library feature be a crate instead".

Reply:
> The request for data is specially prefixed with "If possible" because it is a high bar and I hope makes clear that is not required, but I think suggesting it is useful as a way to at least make people think in concrete terms where possible rather than 'just vibes' type motivations.

I would *love* to see text that attempts to get people thinking in meaningful-to-others terms rather than purely subjective ones. For instance:

- Keep in mind that questions of design can be subjective, and the readers and reviewers may not share your tastes. Try to explain the value of your proposed change in ways meaningful to a broad audience. If you have or can easily get supporting data, provide it (e.g. how widely used a pattern is or how commonly an option is used).